rollup.config.js

```diff
@@ -20,6 +20,7 @@ const typescriptPath = path.resolve(
   __dirname,
   "third_party/node_modules/typescript/lib/typescript.js"
 );
+const gnArgs = fs.readFileSync("args_list.txt", "utf-8").trim();
```

Comment:
@kt3k The looks good except for where you write 'args_list.txt' in 'setup.py'. It think this should be done differently - without using setup.py.

Here's an outline of what I think should happen:

1. I think you should use `exec_script` to generate args_list.txt into a string. Similar to this https://github.com/denoland/deno/blob/33a6409db982d4ee9012cf7eede8f2fb8519f568/BUILD.gn#L135-L138
2. This string should be passed to "main_bundle" here https://github.com/denoland/deno/blob/33a6409db982d4ee9012cf7eede8f2fb8519f568/BUILD.gn#L216-L223 as a new variable.. something like `gn_args = output_from_exec_script`
3. You must then add something to the bundle template, to take "gn_args" and pass it to the rollup program via command line: https://github.com/denoland/deno/blob/33a6409db982d4ee9012cf7eede8f2fb8519f568/libdeno/deno.gni#L4-L31

Reply:
OK. I'll try this approach.

Reply:
Steps 2 and 3 seem feasible, but step 1 doesn't.

When I put `gn args` command in exec_script like:
```gn
deno_build_args = exec_script("third_party/depot_tools/gn.py",
                              [ "args", root_build_dir, "--list", "--short" ],
                              "string")
```
Then it errored like the below:
```console
/Users/kt3k/s/deno/third_party/depot_tools/gn gen /Users/kt3k/s/deno/target/release
ERROR at //BUILD.gn:140:19: Script returned non-zero exit code.
deno_cargo_info = exec_script("third_party/depot_tools/gn.py",
                  ^----------
Current dir: /Users/kt3k/s/deno/target/release/
Command: python /Users/kt3k/s/deno/third_party/depot_tools/gn.py args //target/release --list --short
Returned 1 and printed out:

ERROR Not a build directory.
This command requires an existing build directory. I interpreted your input
"//target/release" as:
  /Users/kt3k/s/deno/target/release/
which doesn't seem to contain a previously-generated build.
```
I think this is because at this stage `gn gen` is not finished and build directory is not ready. And that causes `gn args` to fail. So I think we need to finish the entire `gn gen` before running `gn args`, and that means we can't put `gn args` command in BUILD.gn.

Reply:
Ok, that makes sense, but I think an `action` will work. Here's an (untested) attempt:

```
  action("write_gn_args") {
    script = "//tools/write_gn_args.py"
    args = [ rebase_path(outputs[0], root_build_dir) ]
    outputs = [ "$target_gen_dir/gn_args.txt" ]
  }
```
you'll add `write_gn_args` to the `deps` section of rollup action.

Reply:
Thanks! That worked!